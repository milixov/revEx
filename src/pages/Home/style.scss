@import '@style/variables.scss';

.header {
  height: 120px;
  background: url('/src/assets/background.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position-y: $dimen-xg * 2;
  background-color: black;
  
  padding: $dimen-xg;

  border-radius: 0 0 $dimen-lg $dimen-lg;
  // background-color: $color-background-secondary;

  * {
    color: $color-background;
  }
}

.body {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: $dimen-lg;


  padding: $dimen-xg;
}

.switch {
  $size: 48px;
  width: $size;
  height: $size;
  padding: 0;
  border-radius: $size / 2;
  margin: 0 $dimen-xg;
}
