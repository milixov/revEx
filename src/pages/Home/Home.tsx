import * as React from 'react';

import style from './Home.scss';

interface IHomeProps {
  foo: string;
  bar: string;
}

export const Home = (props: IHomeProps): JSX.Element => (
  <div className={style.sample}>
    Hello from {props.foo} and {props.bar}!
  </div>
);
