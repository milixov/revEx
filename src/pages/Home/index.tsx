import React, { useState } from 'react';

import { initialComponentProps } from '@pages/Home/handlers';
import { Home } from '@pages/Home/Home';

export default (): JSX.Element => {
  const [componentProps] = useState(initialComponentProps);

  return <Home {...componentProps} />;
};
