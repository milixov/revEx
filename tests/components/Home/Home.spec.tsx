import React from 'react';
import renderer from 'react-test-renderer';

import { Home } from '@pages/Home/Home';

describe('Home component', () => {
  test('should render component properly', () => {
    // given
    const dummyComponentProps = {
      bar: 'test',
      foo: 'lorem',
    };

    // when
    const componentRenderer = renderer.create(<Home {...dummyComponentProps} />);
    const tree = componentRenderer.toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
