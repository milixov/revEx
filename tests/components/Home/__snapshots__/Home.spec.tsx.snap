// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render component properly 1`] = `
<div
  className="sample"
>
  Hello from 
  lorem
   and 
  test
  !
</div>
`;
